src/torchcodec/decoders/_core/VideoDecoder.cpp

```diff
@@ -1395,11 +1389,13 @@ void VideoDecoder::convertAudioAVFrameToFrameOutputOnCPU(
 }
 
 UniqueAVFrame VideoDecoder::convertAudioAVFrameSampleFormat(
-    const UniqueAVFrame& avFrame,
+    UniqueAVFrame& avFrame,
     AVSampleFormat sourceSampleFormat,
-    AVSampleFormat desiredSampleFormat
+    AVSampleFormat desiredSampleFormat) {
+  if (sourceSampleFormat == desiredSampleFormat) {
+    return std::move(avFrame);
+  }
```

Comment:
@scotts  WDYT - this is wrong, isn't it? This will end up destructing the `avFrameStream.avFrame` that was passed as a parameter to `convertAudioAVFrameToFrameOutputOnCPU`?

Reply:
On this being wrong - I don't know, that depends on what we need for the calling side. What this will do is take ownership away from the input `avFrame` and give it to the output `UniqueAVFrame`. The question then is: is that okay for `avFrameStream` in the calling context? That is, is it okay for the `avFrameStream` in the calling context to be modified such that it no longer owns the `avFrame`? Because as a consequence of the move, `avFrameStream.avFrame` will become `nullptr`.

And if it's _not_ okay, then should `avFrameStream` in `convertAudioAVFrameToFrameOutputOnCPU()` actually be a const reference?